src/main/java/io/stargate/sgv2/jsonapi/service/embedding/gateway/EmbeddingGatewayClient.java

```diff
@@ -149,9 +151,15 @@ else if (value instanceof Boolean)
             .setEmbeddingRequest(embeddingRequest)
             .setProviderContext(providerContext)
             .build();
-
-    final Uni<EmbeddingGateway.EmbeddingResponse> embeddingResponse =
-        embeddingService.embed(providerEmbedRequest);
+    Uni<EmbeddingGateway.EmbeddingResponse> embeddingResponse;
+    try {
+      embeddingResponse = embeddingService.embed(providerEmbedRequest);
+    } catch (StatusRuntimeException e) {
+      if (e.getStatus().getCode().equals(Status.Code.DEADLINE_EXCEEDED)) {
+        throw ErrorCode.EMBEDDING_PROVIDER_TIMEOUT.toApiException(e, e.getMessage());
+      }
+      throw e;
+    }
```

Comment:
Currently no retry if EGW timed out. Should we add a retry for it?

Reply:
We may eventually need to add one. But let's see how often there are problems -- if you want, you can create an issue now, for future work?

Reply:
Here is the issue: https://github.com/stargate/data-api/issues/1256